Route/CmsPageRouter.php

```diff
@@ -207,7 +210,25 @@ protected function generateFromPage(PageInterface $page, array $parameters = arr
             throw new \RuntimeException(sprintf('Page "%d" has no url or customUrl.', $page->getId()));
         }
 
-        return $this->decorateUrl($url, $parameters, $referenceType);
+        @trigger_error(
+            'The context should implement SiteRequestContextInterface on the next release',
+            E_USER_DEPRECATED
+        );
```

Comment:
Shouldn't this be inside the if statement? When making a BC PR, you should always use the old way by default, but allow the user to use the new, recommended way, so that they can switch to the next major just by fixing all the deprecation notices they see.